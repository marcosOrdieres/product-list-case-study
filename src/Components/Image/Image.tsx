import React from 'react';
import styled from 'styled-components';

const ImageStyled = styled.img`
  width: 100%;
  height: 100%;
  object-fit: cover;
`;

export interface ImageProps {
  src?: string,
};

const Image = (props: ImageProps) => {
  return (
    <ImageStyled src={props.src} />
  )
};

export default Image;