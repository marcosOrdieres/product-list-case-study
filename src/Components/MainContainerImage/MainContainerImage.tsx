import React from 'react';
import styled from 'styled-components';

export interface MainImageContainerProps {
  padding?: boolean,
  imageSrc?: string,
  children?: JSX.Element
};

const MainImageContainerStyled = styled.div<MainImageContainerProps>`
  display: flex;
  width:100px;
  height: 100px;
  background-color:  'white';
  align-items: center;
  justify-content: center;
  padding:${(props) => props.padding ? '10px' : null};
`;

const MainImageContainer = (props: MainImageContainerProps) => {
  return (
    <MainImageContainerStyled padding>
      {props.children}
    </MainImageContainerStyled>
  )
};

export default MainImageContainer;