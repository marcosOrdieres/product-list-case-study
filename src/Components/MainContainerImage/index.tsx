import MainContainerImage from './MainContainerImage';

export default MainContainerImage;
