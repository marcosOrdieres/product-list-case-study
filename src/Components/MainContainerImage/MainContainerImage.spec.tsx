import React from 'react';
import { render } from '@testing-library/react';

import MainContainerImage from './MainContainerImage';
import Image from '../Image';

describe('MainContainerImage', () => {
  it('should render MainContainerImage and match styles', () => {
    const { asFragment } = render(
      <MainContainerImage padding>
        <Image src='' />
      </MainContainerImage>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
