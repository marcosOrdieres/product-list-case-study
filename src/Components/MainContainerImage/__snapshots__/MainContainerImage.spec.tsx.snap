// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContainerImage should render MainContainerImage and match styles 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBQB kjmAOv"
  >
    <img
      class="sc-gsTEea kWaMru"
      src=""
    />
  </div>
</DocumentFragment>
`;
